### ISSUE TITLE
Alias Calls Not Working

### ISSUE THREAD

**Issue by iamnafets** [PostHog/posthog#23609]
## Bug description

I'm a paying user of Posthog cloud. I create distinct IDs on my side and initialize the posthog client with them. When the user logs in, I alias those distinct ids to the users id. This user has a record of an alias call, no merge restrictions, and isn't merged into the user:

https://us.posthog.com/project/55874/person/bnihzbyfgjtywivs16qlx5t6

I'm 60% sure this was actually previously working this year.

## How to reproduce

1. Create a random distinct ID `d`, initialize the posthog client with it.
2. `capture` some events
3.  Run `alias({ distinctId: d, alias: "some_user" })`
4. Observe that `d` is not associated with `some_user`.

### Debug info

```shell
- [x] PostHog Cloud, Debug information: 
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/01909d41-cc76-7e89-a8f1-fb3c701363e9?t=3174 
Admin: http://go/adminOrgUS/0188c19e-3b1a-0000-6aa1-d006cbca5583 (Project: 55874)
Sentry: http://go/sentryUS/55874
```

**Comment by MarconLP** [comment]
Hey @iamnafets, could you share a bit more context about the first step? How did you initialize PostHog with the distinct_id `d`? Has one of the distinct_id's been used before to capture events already?

**Comment by iamnafets** [comment]
Hey @MarconLP , here's a sample call:

```
        posthog.init("SECRET", {
            api_host: 'https://app.posthog.com',
            disable_session_recording: true,
            bootstrap: {
                distinctID: distinctId, featureFlags: flagData
            }
        });
```
In this example, `some_user` has been used before (it's the authenticated user id of the user), whereas `d` is brand new. Per the documentation, this shouldn't be a problem (i.e. merge restrictions).

**Comment by bretthoerner** [comment]
Hey @iamnafets, can you swap the `distinctId` and `alias` arguments in your `alias()` call?

Providing the existing/identified user as the "alias" argument isn't allowed by merge rules. If `some_user` existed and has been merged into before, then you want to use it as the `distinctId` and add an alias of `your-random-distinct-id-that-hasnt-been-merged-before`

**Comment by iamnafets** [comment]
@bretthoerner Giving that a try, thanks. The documentation might be a bit clearer here.

**Comment by bretthoerner** [comment]
> @bretthoerner Giving that a try, thanks. The documentation might be a bit clearer here.

Yes, definitely, I meant to say that and forgot 😅. Let me know if that works for you and then I'll work to improve the docs.
